setup.py

```diff
@@ -159,11 +159,16 @@ def run(self):
         'markdown>=3.3.0',
         'matplotlib>=3.0.0',
         'natsort>=7.1.1',
-        'networkx>=1.11',
-        'numpy>=1.19.2',
+        'networkx<=2.5;python_version=="3.6"',
+        'networkx<=2.6;python_version=="3.7"',
+        'networkx>=2.7;python_version>"3.7"',
+        'numpy<1.20.0;python_version=="3.6"',
+        'numpy>=1.20.3;python_version>="3.7"',
         'omegaconf==2.0.6',
         'openpyxl>=3.0.7',
-        'pandas==1.1.5',
+        'pandas<1.2.0;python_version=="3.6"',
+        'pandas<1.4.0;python_version=="3.7"',
+        'pandas>=1.4.0;python_version>"3.7"',
```

Comment:
> networkx >= 2.5
numpy>=1.19.2
pandas>=1.1.5

will these conditions work directly? setuptools will resolve them automatically as long as there's no conflict?

Reply:
if it's the ideal case that there's a clean env that's ok, but sometimes the lcoal env may pre-installed some packages, and we use --system-site-packages to create venv, so if local env pre-isntalled 1.19.2 numpy while the latest pandas installed when installing superbench, it will cause error

Reply:
> if local env pre-isntalled 1.19.2 numpy while the latest pandas installed

I think this means before installing superbench package the existing python packages already have conflict?
Maybe we can check the return code of [`python -m pip check`](https://pip.pypa.io/en/stable/cli/pip_check/) in the beginning of setup.py? It's hard to cover all possible conflicts (e.g., specify python conditions for all packages) so better to rely on setuptools/pip.

Here's an example output of your proposed case:
```
# pip check
pandas 1.5.3 has requirement numpy>=1.20.3; python_version < "3.10", but you have numpy 1.19.2.
# echo $?
1
```

---

Or can we remove `--system-site-packages` in the document? start from clean env sounds better and can avoid conflicts with apt packages as well.